correlation-parser/src/timer.rs

```diff
@@ -45,10 +46,7 @@ impl<E, T> Timer<E, T> for Watchdog where E: Event + Send, T: Template<Event=E>
                         Err(TryRecvError::Empty) => (),
                     }
 
-                    match correlator.lock() {
-                        Ok(mut guard) => guard.elapse_time(delta),
-                        Err(_) => break
-                    }
+                    cb();
```

Comment:
The parking and not parking states have similar parts. They should be factored out into a separated function, and move outside from the if/else statement